src/core_plugins/tile_map/public/maps_visualization.js

```diff
@@ -181,12 +182,13 @@ export function MapsVisualizationProvider(serviceSettings, Notifier, getAppState
       const { minZoom, maxZoom } = this._getMinMaxZoom();
 
       if (mapParams.wms.enabled) {
-        //Switch to WMS
+        // Switch to WMS
         if (maxZoom > this._kibanaMap.getMaxZoomLevel()) {
           //need to recreate the map with less restrictive zoom
+          this._kibanaMap.removeLayer(this._geohashLayer);
           this._geohashLayer = null;
-          this._kibanaMapReady = this._makeKibanaMap();
```

Comment:
hmm ... i got lost a bit here, why we no longer recreate the map ?

Reply:
wasn't the only change to tile map just extracting the wms options into separate directive ?

Reply:
because we upgraded leaflet to 1.0, we now have setters for the zoom-level. before min/max zoom had  to be set at construction-time and couldn't be altered. 

this refactors this so we don't recreate the map when we swap the baselayer (tms<-->wms)